### PR TITLE
Add com.powerball253.eternalmodmanager

### PR DESCRIPTION
**Pull request by PowerBall253**
https://github.com/PowerBall253/EternalModManager

- [X] I have read the [App Requirements](https://github.com/flathub/flathub/wiki/App-Requirements) and [App Maintenance](https://github.com/flathub/flathub/wiki/App-Maintenance) pages.
- [X] My pull request follows the instructions at [App Submission](https://github.com/flathub/flathub/wiki/App-Submission).
- [ ] I am using only the minimal set of permissions.
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project.
- [X] I own the domain used in the application ID.
- [X] Any additional patches or files have been submitted to the upstream projects concerned.

Non standard permissions used:
```
--filesystem=host
```
As this program needs to operate in a game directory, that could be in a lot of places in the filesystem, it requires access to the user's files. I'll work on removing this once Electron 14 comes out, which will introduce the GtkFileChooserNative portal, allowing me to avoid having to access the `host` filesystem.

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build com.powerball253.eternalmodmanager

**Comment by PowerBall253** [comment]
bot, build com.powerball253.eternalmodmanager

**Comment by PowerBall253** [comment]
bot, build com.powerball253.eternalmodmanager

**Comment by PowerBall253** [comment]
bot, build com.powerball253.eternalmodmanager

**Comment by PowerBall253** [comment]
Latest commit solves all flatpak-related problems, will wait for review.

**Comment by PowerBall253** [comment]
bot, build com.powerball253.eternalmodmanager

**Comment by PowerBall253** [comment]
bot, build com.powerball253.eternalmodmanager

**Comment by PowerBall253** [comment]
bot, build com.powerball253.eternalmodmanager

**Comment by PowerBall253** [comment]
I have fixed all flatpak-related issues in my app, I will wait for review.

File: com.powerball253.eternalmodmanager.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  - org.freedesktop.Sdk.Extension.node14
+command: run.sh
+separate-locales: false
+finish-args:
+  # X11 stuff
+  - --share=ipc
+  - --socket=x11
+  # Needs host fs access due to Steam permissive library paths
+  # Should no longer be needed when Electron 14 becomes stable, introducing the GtkNativeFileChooser portal
+  - --filesystem=host
```
Comment (barthalion) [inline com.powerball253.eternalmodmanager.yml:17]: Wouldn't `~/.var/app/com.valvesoftware.Steam` and `~/.local/share/Steam` be good enough then?

File: com.powerball253.eternalmodmanager.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  # Networking to download the modding tools if needed
+  - --share=network
+build-options:
+  append-path: /usr/lib/sdk/node14/bin
+  cflags: -O2 -g
+  cxxflags: -O2 -g
```
Comment (barthalion) [inline com.powerball253.eternalmodmanager.yml:23]:
```suggestion
```

File: com.powerball253.eternalmodmanager.yml (hunk 1 above)
Comment (PowerBall253) [reply com.powerball253.eternalmodmanager.yml:17]: There's also other common paths for libraries like `/media`, `/run/media` and `/mnt`, is there a way I can gain access to this paths too? Maybe use something like
```
- --filesystem=home
- --filesystem=/media
- --filesystem=/mnt
- --filesystem=/run/media
```

**Comment by PowerBall253** [comment]
bot, build com.powerball253.eternalmodmanager

**Comment by nedrichards** [comment]
bot, build com.powerball253.eternalmodmanager

**Pull request closed without merging**